File: backend/app/schemas/analysis.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.domain.enums import AnalysisCategory, Sentiment


class AnalysisOutput(BaseModel):
    """Schema for validating LLM output."""

    sentiment: Sentiment
    category: AnalysisCategory
    urgency: int = Field(ge=1, le=5)
    themes: list[str] = Field(min_length=1, max_length=5)
    suggested_action: str = Field(min_length=1)
    summary: str = Field(min_length=1)


class AnalysisResponse(BaseModel):
    id: int
    sentiment: Sentiment
    category: AnalysisCategory
    urgency: int
    themes: list[str]
    suggested_action: str
    summary: str
    model: str
    tokens_used: int
    latency_ms: int
    cost_cents: float
    is_valid: bool
    validation_errors: list[str] | None
    created_at: datetime

File: frontend/src/schemas/analysis.ts
import { z } from 'zod'
import { analysisCategoryEnum, sentimentEnum } from './feedback'

export const analysisResponseSchema = z.object({
  id: z.number(),
  sentiment: sentimentEnum,
  category: analysisCategoryEnum,
  urgency: z.number().min(1).max(5),
  themes: z.array(z.string()),
  suggested_action: z.string(),
  summary: z.string(),
  model: z.string(),
  tokens_used: z.number(),
  latency_ms: z.number(),
  cost_cents: z.number(),
  is_valid: z.boolean(),
  validation_errors: z.array(z.string()).nullable(),
  created_at: z.string(),
})

export type AnalysisResponse = z.infer<typeof analysisResponseSchema>
